I think `value.text` may also be None:

I think `value.text` may also be None:
```
>>> err = SyntaxError('Interface magics have no options, got ')
>>> err
SyntaxError('Interface magics have no options, got ')
>>> err.msg
'Interface magics have no options, got '
>>> err.text
>>> type(err.text)
<class 'NoneType'>
```

at least with the latest version we now get:
```
ile ~/miniconda3/envs/sage-dev/lib/python3.12/site-packages/IPython/core/interactiveshell.py:3350, in InteractiveShell.run_cell_async(self, raw_cell, store_history, silent, shell_futures, transformed_cell, preprocessing_exc_tuple, cell_id)
   3348 # Display the exception if input processing failed.
   3349 if preprocessing_exc_tuple is not None:
-> 3350     self.showtraceback(preprocessing_exc_tuple)
   3351     if store_history:
   3352         self.execution_count += 1

File ~/miniconda3/envs/sage-dev/lib/python3.12/site-packages/IPython/core/interactiveshell.py:2164, in InteractiveShell.showtraceback(self, exc_tuple, filename, tb_offset, exception_only, running_compiled_code)
   2159     return
   2161 if issubclass(etype, SyntaxError):
   2162     # Though this won't be called by syntax errors in the input
   2163     # line, there may be SyntaxError cases with imported code.
-> 2164     self.showsyntaxerror(filename, running_compiled_code)
   2165 elif etype is UsageError:
   2166     self.show_usage_error(value)

File ~/miniconda3/envs/sage-dev/lib/python3.12/site-packages/IPython/core/interactiveshell.py:2254, in InteractiveShell.showsyntaxerror(self, filename, running_compiled_code)
   2252 # If the error occurred when executing compiled code, we should provide full stacktrace.
   2253 elist = traceback.extract_tb(last_traceback) if running_compiled_code else []
-> 2254 stb = self.SyntaxTB.structured_traceback(etype, value, elist)
   2255 self._showtraceback(etype, value, stb)

File ~/miniconda3/envs/sage-dev/lib/python3.12/site-packages/IPython/core/ultratb.py:1243, in SyntaxTB.structured_traceback(self, etype, evalue, etb, tb_offset, context)
   1241         value.text = newtext
   1242 self.last_syntax_error = value
-> 1243 return super(SyntaxTB, self).structured_traceback(
   1244     etype, value, etb, tb_offset=tb_offset, context=context
   1245 )

File ~/miniconda3/envs/sage-dev/lib/python3.12/site-packages/IPython/core/ultratb.py:212, in ListTB.structured_traceback(self, etype, evalue, etb, tb_offset, context)
    210     out_list.extend(self._format_list(elist))
    211 # The exception info should be a single entry in the list.
--> 212 lines = "".join(self._format_exception_only(etype, evalue))
    213 out_list.append(lines)
    215 # Find chained exceptions if we have a traceback (not for exception-only mode)

File ~/miniconda3/envs/sage-dev/lib/python3.12/site-packages/IPython/core/ultratb.py:341, in ListTB._format_exception_only(self, etype, value)
    327 output_list.append(
    328     theme_table[self._theme_name].format(
    329         [(Token, "  ")]
   (...)    336     )
    337 )
    338 if textline == "":
    339     # sep 2025:
    340     # textline = py3compat.cast_unicode(value.text, "utf-8")
--> 341     assert isinstance(value.text, str)
    342     textline = value.text
    344 if textline is not None:

AssertionError: 

    shell.run_cell(bad_syntax)
```

_Originally posted by @tobiasdiez in https://github.com/ipython/ipython/pull/14995#discussion_r2404428120_

--- a/IPython/core/ultratb.py
+++ b/IPython/core/ultratb.py
@@ -338,8 +338,11 @@ class ListTB(TBTools):
                 if textline == "":
                     # sep 2025:
                     # textline = py3compat.cast_unicode(value.text, "utf-8")
-                    assert isinstance(value.text, str)
-                    textline = value.text
+                    if value.text is None:
+                        textline = ""
+                    else:
+                        assert isinstance(value.text, str)
+                        textline = value.text
 
                 if textline is not None:
                     i = 0

--- a/tests/test_ultratb.py
+++ b/tests/test_ultratb.py
@@ -498,3 +498,10 @@ def test_handlers():
     except:
         handler(*sys.exc_info())
     buff.write("")
+
+
+def testSyntaxError():
+    cell = "raise SyntaxError()"
+    expected = "SyntaxError\n"
+    with tt.AssertPrints(expected):
+        ip.run_cell(cell)

